apps/avifenc.c

```diff
@@ -727,6 +727,18 @@ int main(int argc, char * argv[])
     image->yuvRange = requestedRange;
     image->alphaPremultiplied = premultiplyAlpha;
 
+    if ((image->matrixCoefficients == AVIF_MATRIX_COEFFICIENTS_IDENTITY) && (image->yuvFormat != AVIF_PIXEL_FORMAT_YUV444)) {
+        // matrixCoefficients was likely set to AVIF_MATRIX_COEFFICIENTS_IDENTITY as a side effect
+        // of --lossless, and Identity is only valid with YUV444. Set this back to the default.
+        image->matrixCoefficients = AVIF_MATRIX_COEFFICIENTS_BT601;
+
+        if (cicpExplicitlySet) {
+            // Only warn if someone explicitly asked for identity.
+            printf("WARNING: matrixCoefficients may not be set to identity (0) when subsampling. Resetting MC to defaults (%d).\n",
+                   image->matrixCoefficients);
+        }
+    }
+
```

Comment:
IMPORTANT: This change is incorrect. 'image' is just created by an avifImageCreateEmpty() call at line 721. So at line 730, image->yuvFormat is still the default value set by the avifImageCreateEmpty() call.

Reply:
Done
I changed to check `input.requestedFormat`, which holds the value specified by user. However y4m input doesn't use this setting so we have to check again. There's still the case that a yuv444 y4m input with `--cicp 1/13/0 -y 420` will make avifenc wrongly change the MC, but as the user is doing wrong things I think it's OK.